Update the legend item properties (#4304)
qgis/QGIS-Documentation: source/docs/user_manual/print_composer/composer_items/composer_legend.rst

In Main properties you can:

* Change the :guilabel:`Title` of the legend. It can be made dynamic using the
  :ref:`data-defined override <data_defined>` setting, useful for example when

>   generating atlas;

Reviewer: ```suggestion
  generating an atlas;
```

* Choose which :guilabel:`Map` item the current legend will refer to.

>   By default, the map over which the legend item is drawn is picked and if none,

Reviewer: ```suggestion
  By default, the map over which the legend item is drawn is picked. If none,
```

  then it falls back to the `reference map <reference_map>`.
* Wrap the text of the legend on a given character: each time the character
  appears, it's replaced with a line break;
* Set the symbols and text placement in the legend: the :guilabel:`Arrangement`
  can be :guilabel:`Symbols on left` or :guilabel:`Symbols on right`. The default

>   value depends on whether the locale in use is right-to-left based or not.

Reviewer: ```suggestion
  value depends on the locale in use (right-to-left based or not).
```

    features are nevertheless kept and shown in the layout map item.
  * |addExpression| :sup:`Add symbol expression` allows you to add expressions

>     to each symbol label of a given layer. The label is represented by 

Author: ```suggestion
    to each symbol label of a given layer. New variables (``@symbol_label``,
    ``@symbol_id`` and ``@symbol_count``) help you interact with the legend
     entry.
```

>     the **@symbol_label** variable in the expression.

Author: Actually I was tempted to remove that mention but I can expand to the others. 
How would you reuse the @symbol_id in an example?

>     the **@symbol_label** variable in the expression.

Author: ```suggestion
```

>     the **@symbol_label** variable in the expression.

Reviewer 2: Yes it's pretty advanced stuff... Feel free to remove it.

    
    For example, given a categorized layer, you can append to each class in the
    legend their number of features, ie *class (number)*:
    
    #. Select the layer entry in the legend tree
    #. Press the |addExpression| :sup:`Add symbol expression` button, opening
       the :guilabel:`Expression String Builder` dialog
    #. Enter the following expression::

        concat( @symbol_label, ' (', @symbol_count, ')' )

    #. Press :guilabel:`OK`

> 

Author: > well something to talk in another place

Few days ago, I think you suggested we discuss something also in another place but I can't remember what and I'm not sure we'll do until the "issue" arises again... and if there still are docs...

> 

Reviewer 2: I am trying to lead the discussion now... I know you are tired of so much talking and no action 😔

> 

Author: :see_no_evil: :hear_no_evil: :speak_no_evil:

  |checkbox| :guilabel:`Only show items inside current atlas feature` option.

Fonts
-----

The :guilabel:`Fonts` group of the legend :guilabel:`Item Properties` panel
provides the following functionalities:

.. _figure_layout_legend_fonts:

.. figure:: img/legend_fonts.png
   :align: center

   Legend Fonts properties

* You can change the font of the legend title, group, subgroup and item (feature)
  in the legend item using the :ref:`font selector <font_selector>` widget

> * You can set for each of these levels the text :guilabel:`Alignment`: it can

Reviewer: ```suggestion
* For each of these levels you can set the text :guilabel:`Alignment`: it can
```

  be :guilabel:`Left` (default for left-to-right based locales),
  :guilabel:`Center` or :guilabel:`Right` (default for right-to-left based
  locales).
* You set the :guilabel:`Color` of the labels using the :ref:`color selector
  <color-selector>` widget. The selected color will apply to all the font items
  in the legend.

Columns
-------

Under the :guilabel:`Columns` group of the legend :guilabel:`Item Properties`

> panel, Legend items can be arranged over several columns:

Reviewer: ```suggestion
panel, legend items can be arranged over several columns:
```
